### DISCUSSION TITLE
Avoid codemagic.yaml from being overriden by branches other than main

### DISCUSSION THREAD

**Discussion by JCharlesAN** [codemagic-ci-cd/codemagic-docs#2082] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [x] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Android (Kotlin)

### Steps to reproduce

I have set all my deployment and build configuration for production in codemagic in the main branch codemagic.yaml. I want this to be the only way to modify the CI/CD configuration. My main branch is push protected and can only be modified by few people.
I have set an exclusion pattern for the triggers of my workflow "!(main)"

### Expected results

Even if i create a codemagic.yaml in a test branch, automatic build should not be triggered thanks to the exclusion pattern

### Actual results

If I include 'test' as an inclusion pattern in a newly created codemagic.yaml file in the test branch, autobuild will get triggered with the test branch CI/CD configs, potentially allowing the creation of a signed app with unwanted modifications form anyone with push access to any branch.

**Comment by dtrdic** [comment]
Hello @JCharlesAN, correct me if i understood you wrongly.. do you actually want the build to be triggered whenever someone merges a PR into your main branch?

**Reply by JCharlesAN** [reply]
Hello @dtrdic thanks for answering. My final goal would be:
1. Have a set of branches which will trigger autobuild only by pushing to those branches (no pr needed)
2. Have all the options defined in the main branch codemagic.yml (which will be locked for push except for few collaborators)
3. Be sure that a collaborator will not be able to push a new codemagic.yml into an unauthorized branch and trigger the build by adding the new branch into the triggering section (basically ignoring the config in point 2 and opening the door for possible unauthorized scripts which the new codemagic.yml might contain)

**Reply by dtrdic** [reply]
Hello @JCharlesAN.. so,
1. For the branches that should trigger autobuild without requiring a pull request, you can set up the triggering options like this:
```
      triggering:
        events:
           - push
        branch_patterns:
        - pattern: "dev_branch" //Replace with the desired branch name
          include: true
          source: false
```
2. This unfortunately can't be limited by users(collaborators) but you can specify the build triggers like this to ensure the build only runs when branches are merged into the main branch, so the triggering section of your main branch would look like this:
```
      triggering:
        events:
            - pull_request
        branch_patterns:
        - pattern: "main"
          include: true
          source: false
```
so, with Pull Request trigger, you have the ability to control and authorize code changes, but there's no automated way to limit specific users from triggering builds.

Also, regarding this "_Even if i create a codemagic.yaml in a test branch, automatic build should not be triggered thanks to the exclusion pattern_".. if you don't want the build to be triggered on the branch creation uncheck this webhook event, in your github repository
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/a300f87e-2192-4f05-82e3-b76429f8ae49)

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.
